Source/Base/Base/Math/Spline.cpp

```diff
@@ -0,0 +1,261 @@
+        _points.erase(it);
+        _storage.MarkAsDirty();
+    }
+
+    void Spline::SetStep(u32 newStep)
+    {
+        newStep = Math::Max((i32)newStep, 1);
+
+        if (_step == newStep)
+            return;
+
+        _step = newStep;
+        _storage.SetStep(newStep);
+    }
+
+    void Spline::SetType(InterpolationType newType)
+    {
+        if (newType < InterpolationType::None || newType >= InterpolationType::MAX_TYPE || _type == newType)
+            return;
+
+        _type = newType;
+        _storage.MarkAsDirty();
+    }
+
+    void Spline::SetAlpha(f32 newAlpha)
+    {
+        _alpha = Math::Clamp(newAlpha, 0.f, 1.f);
+        _storage.MarkAsDirty();
+    }
+
+    bool Spline::DoInterpolation()
+    {
+        if (!_storage.IsDirty())
+            return false;
+
+        // need to have minimum 2 points
+        if (_points.size() < 2 || _step == 0)
+            return false;
+
+        _storage.Clear();
+        f32 tStep = (1.f) / static_cast<f32>(_step);
+        if (IsInterpolatedWithControl())
+        {
+            for (i32 n = 0; n < _points.size() - 1; n++)
+            {
+                SplinePoint currentPoint = _points[n];
+                SplinePoint nextPoint = _points[n + 1];
+
+                std::vector<vec3> currentPortion;
+                currentPortion.reserve(_step + 1);
+
+                for (u32 i = 0; i < _step + 1; i ++)
+                {
+                    f32 t = tStep * static_cast<f32>(i);
+                    currentPortion.emplace_back(InterpolateControl(t, currentPoint, nextPoint));
+                }
+
+                _storage.AddPortion(currentPortion);
+            }
+        }
+        else
+        {
+            vec3* ptr = GetPtrPoints();
+            for (i32 n = 0; n < (_points.size() * 3) - 4; n++)
+            {
+                std::vector<vec3> currentPortion;
+                currentPortion.reserve(_step + 1);
+
+                for (u32 i = 0; i < _step + 1; i++)
+                {
+                    f32 t = tStep * static_cast<f32>(i);
+                    currentPortion.emplace_back(InterpolateCombined(t, ptr, n, _alpha));
+                }
+
+                _storage.AddPortion(currentPortion);
+            }
+        }
+
+        return true;
+    }
+
+    vec3* Spline::GetPtrPoints()
+    {
+        if (_points.empty())
+            return nullptr;
+
+        vec3* ptr = new vec3[_points.size() * 3];
+        for (i32 i = 0; i < _points.size(); i++)
+        {
+            std::memcpy(ptr + (3 * i), _points[i].combined.data, 3 * sizeof(vec3));
+        }
+
+        return ptr;
+    }
+
+    bool Spline::IsInterpolatedWithControl()
+    {
+        switch (_type)
+        {
+        case InterpolationType::BSpline:
+        case InterpolationType::CatmullRom:
+        case InterpolationType::CatmullRom_Uniform:
+        case InterpolationType::CatmullRom_Centripetal:
+        case InterpolationType::CatmullRom_Chordal:
+            return false;
```

Comment:
Please add a space between the different cases (Please fix this for all switch cases in this file)